site/zenodo_rdm/migrator/transform/records.py

```diff
@@ -0,0 +1,206 @@
+            for c in data:
+                r = {"person_or_org": {"type": "personal"}}
+                if c.get("affiliation"):
+                    r["affiliations"] = [{"name": c["affiliation"]}]
+                if c.get("orcid"):
+                    r["person_or_org"]["identifiers"] = [
+                        {"scheme": "orcid", "identifier": c["orcid"]},
+                    ]
+                name = HumanName(c["name"])
+                r["person_or_org"]["given_name"] = name.first
+                r["person_or_org"]["family_name"] = name.surnames
+                # autocompleted by RDM Metadata schema
+                r["person_or_org"]["name"] = f"{name.surnames}, {name.first}"
+
+                ret.append(r)
+            return ret
+
+        def _resource_type(data):
+            t = data["type"]
+            st = data.get("subtype")
+            return {"id": f"{t}-{st}"} if st else {"id": t}
+
+        record = entry["json"]
+        r = {
+            "title": record["title"],
+            "description": record["description"],
+            "publication_date": record["publication_date"],
+            "resource_type": _resource_type(record["resource_type"]),
+            "creators": _creators(record["creators"]),
+        }
+
+        return r
+
+    def _custom_fields(self, entry):
+        """Transform custom fields."""
+        metadata = entry.get("json", {})
+        cf = {
+            "journal:journal": {
+                "title": metadata.get("journal", {}).get("title"),
+                "issue": metadata.get("journal", {}).get("issue"),
+                "pages": metadata.get("journal", {}).get("pages"),
+                "volume": metadata.get("journal", {}).get("volume"),
+                "issn": metadata.get("journal", {}).get("issn"),
+            },
+            "meeting:meeting": {
+                "acronym": metadata.get("meeting", {}).get("acronym"),
+                "dates": metadata.get("meeting", {}).get("dates"),
+                "place": metadata.get("meeting", {}).get("place"),
+                "session_part": metadata.get("meeting", {}).get("session_part"),
+                "session": metadata.get("meeting", {}).get("session"),
+                "title": metadata.get("meeting", {}).get("title"),
+                "url": metadata.get("meeting", {}).get("url"),
+            },
+            "imprint:imprint": {
+                "isbn": metadata.get("imprint", {}).get("isbn"),
+                "publisher": metadata.get("imprint", {}).get("publisher"),
+                "place": metadata.get("imprint", {}).get("place"),
+                # QUESTION: the following two are not found in Zenodo
+                # "title": metadata.get("journal", {}).get("title"),
+                # "pages": metadata.get("journal", {}).get("volume"),
+            },
+            "thesis:university": metadata.get("thesis", {}).get("university"),
+            # QUESTION: migrate Zenodo thesis.supervisors as record collaborators?
```

Comment:
@slint @zzacharo What do we do about Zenodo's `thesis.supervisors`. They do not map to any new field, however we could move them to collaborators (not sure which role)

Reply:
We removed the supervisors from the thesis field for now, given that `supervisors` can be retrieved from the contributors whose role is `Supervisor`. However, that supervisor would only be the thesis supervisor if the resource type is `Publication/Thesis`. But correct me if I'm, wrong  @slint

Reply:
@alejandromumo I agree with that, **new** records in ZenodoRDM will have supervisors in the contributors field. However, my comment was on the reverse order about **migrating** the old Zenodo ones (that are already in `thesis.supervisors`), which the zenodo_rdm.migrator currently simply discards (i.e. we are loosing metadata).

Reply:
So, the rule is that we'll just migrate these into the `contributors` field (which is not yet implemented in this class, but has to).

Something to keep in mind is that we should keep the legacy REST API and the migrated metadata in sync, i.e. if someone submits `thesis.supervisors` via the legacy REST API, it should end up in `contributors`

Reply:
Perfect, I've added a light version of contributors migration. For the moment only takes care of adding the supervisors (which is what concerns this PR) but sets the logistics to allow adding contributors already.